crypto/sm2/sm2_sign.c

```diff
@@ -11,6 +11,7 @@
 
 #include "internal/sm2.h"
 #include "internal/sm2err.h"
+#include "../ec/ec_lcl.h" /* ec_group_do_inverse_ord() */
```

Comment:
@mattcaswell is this the right way to include stuff from the private API of the EC submodule?

Reply:
No...if there are functions shared between internal sub-modules then they should be put in a header file in crypto/include/internal. You might have to create a new one if it doesn't already exist.

Reply:
@mattcaswell So probably this latter change needs to be applied to #6116 before merging?

Reply:
I mean, including a definition of `ec_group_do_inverse_ord()` in a header in `crypto/include/internal`

Reply:
Well, you could do it either way. In some ways it makes more sense to do it here, i.e. don't make it into an internal include file until you actually need to use it. But whatever you think is easiest.

Reply:
Right, concern here is that relative path in #include can vary between implementations. To counter specifically this concern one can add `INCLUDE[sm2_sign.o]=../ec` to crypto/sm2/build.info and include ec_lcl.h.

Reply:
> Right, concern here is that relative path in #include can vary between implementations.

That is *a* concern, but it is not my main concern. As a general principle we should not expose all of the internals of one sub-module to other sub-modules. There should be a clearly defined "API" between one sub-module and another. The internal include files define what it is reasonable for other sub-modules to have access to, and what it is not.

For that reason I do not think we should take that approach.

Reply:
Ok, so let's get #6116 merged, given that it's ready, and then I'll move the prototype declaration where is more appropriate in this PR!

Thanks for the feedback!